cypress/plugins/testDataProcessor.md

```diff
@@ -0,0 +1,233 @@
+
+Note:
+ - When data is not present in the fixtures, returning the passed the key as is.
+ - `Mock Firebolt` is the default device and mac addressed is not required.
+   
+   
+
+Ex:
+
+```
+request: testDataHandler("params","Account_Id")
+```
+```
+response: {value: 123456}
+```
+```
+request: testDataHandler("content","Device_ID")
+```
+```
+response: 1234
+```
+```
+request: testDataHandler("params","CYPRESSENV-capabilitiesList")
+```
+```
+response: "CYPRESSENV-capabilitiesList"
+```
+
+
++++++++++++++++++++++++++
+
+### > testDataParser
+Fetching the data from json files based on the priority as shown below
+  - External <module>.json from configModule (If applicable)
+  - Internal <module>.json from fixtures (If applicable)
+  - default.json
+   
+##### Params:
+| Param | Definition|
+| --- | --- |
+| requestType | Type of request. param or content |
+| dataIdentifier | Key to be used to fetch param or content data from the fixtures |
+
+Ex:
+
+```
+request: testDataParser('TRUE', 'params');
+```
+```
+response: true
+```
+```
+request: testDataParser('ACCESSIBILITY_FONTFAMILY_MONOSPACE', 'params');
+```
+```
+response: {"value": "monospaced_sanserif"}
+```
+
+### > fetchAndParseDataFromJSON
+Function used to fetch the data from the file and parse the data based on passed key.
+   
+##### Params:
+| Param | Definition|
+| --- | --- |
+| filePath | Type of request. param or content |
+| dataIdentifier | Key to be used to fetch data from the file. |
+| requestType | request type contains params, context or content. |
+
+Ex:
+
+```
+fetchAndParseDataFromJSON('./example.json', 'abc' , 'params')
+```
+### > combineValidationObjectsJson
+Function to combine all validation JSON files from FCS and config module.
+
+Ex:
+
+```
+combineValidationObjectsJson()
+```
+
+### > extractModuleName
+
+Parsing the module name from the dataIdentifier passed.
+
+##### Params:
+| Param | Definition|
+| --- | --- |
+| dataIdentifier | Key to be used to extranct the module name. |
+
+Ex:
+
+```
+request: extractModuleName("ACCESSIBILITY_CLOSEDCAPTIONS_TRUE")
+```
+```
+response: accessibility
+```
+
+### > parseDataFromJson
+
+Function to fetch the data from the passed JSON.
+
+##### Params:
+| Param | Definition|
+| --- | --- |
+| data | JSON data needed to parse based on key. |
+| dataIdentifier | Key to be used to find value from JSON. |
+| requestType | request type contains params, context or content. |
+
+Ex:
+
+```
+request: parseDataFromJson({"ACCESSIBILITY_CLOSEDCAPTIONS_TRUE":"true"},"ACCESSIBILITY_CLOSEDCAPTIONS_TRUE", "Params")
+```
+```
+response: true
+```
+
+### > fetchDataFromFile
+
+Function to fetch the data from the passed JSON.
+
+##### Params:
+| Param | Definition|
+| --- | --- |
+| filePath | filePath |
+
+Ex:
+
+```
+fetchDataFromFile('./filepath.json')
+```
+
+### > mergeJsonFilesData
+
+Function to fetch the data from the passed JSON.
+
+##### Params:
+| Param | Definition|
+| --- | --- |
+| paths | array of files path. |
+
+Ex:
```

Comment:
mergeJsonFilesData function description and readme doesnt match . 
mergeJsonFilesData is used to fetch the data from the passed JSON ???

Reply:
Corrected the description. mergeJsonFilesData will read and merge the data from the passed array of file paths.